### ISSUE TITLE
Missing projection metadata for some geopackage layers exported or processed without changing the CRS

### ISSUE THREAD

**Issue by josephholler** [qgis/QGIS#34614]
Here's a *fun* one:

If a process saves an output from a geopackage layer without changing its projection (e.g. through exporting features, or through running an algorithm like centroids), the output appears with an undefined projection if the input CRS was EPSG `3653` (NAD83(NSRS2007) / Rhode Island). 

If the output is a shapefile, the .prj file contains only this:
`GEOGCS["GCS_WGS_1984",DATUM["D_WGS_1984",SPHEROID["WGS_1984",6378137.0,298.257223563]],PRIMEM["Greenwich",0.0],UNIT["Degree",0.0174532925199433]]`

If the output is a geopackage layer, the SRID of geometry is `0`, and the SRID values recorded for the layer in the `gpkg_contents` and `gpkg_geometry_columns` tables is also `0`.

Interestingly, layers with EPSG code `6567` (NAD83(2011) / Rhode Island), behave normally. The only difference between the two projections that I can discern is that `3653` includes `+towgs84=0,0,0,0,0,0,0` in its Proj4 code.

If you export a layer with epsg code `3653` and change the CRS while exporting, then the output contains a correctly defined CRS. It's only when the output has the same CRS as the input that the CRS is lost.

The attached geopackage contains two layers, one with each of the SRID's discussed.

[ri_state_plane_kerfuffle.zip](https://github.com/qgis/QGIS/files/4242374/ri_state_plane_kerfuffle.zip)

**QGIS and OS versions**

QGIS version | 3.11.0-Master | QGIS code revision | 7c7c2d775b
-- | -- | -- | --
Compiled against Qt | 5.11.2 | Running against Qt | 5.11.2
Compiled against GDAL/OGR | 3.1.0dev | Running against GDAL/OGR | 3.1.0dev
Compiled against GEOS | 3.8.0-CAPI-1.13.1 | Running against GEOS | 3.8.0-CAPI-1.13.1
Compiled against SQLite | 3.29.0 | Running against SQLite | 3.29.0
PostgreSQL Client Version | 11.5 | SpatiaLite Version | 4.3.0
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.8
Compiled against PROJ | 7.0.0 | Running against PROJ | Rel. 7.0.0, March 1st, 2020
OS Version | Windows 10 (10.0) | This copy of QGIS writes debugging output.
Active python plugins | db_manager; MetaSearch; processing

**Comment by nyalldawson** [comment]
Please test with 3.12 final

**Comment by josephholler** [comment]
Does a compiled version of 3.12 exist?

**Comment by josephholler** [comment]
Ok, confirmed the issue still exists on 3.12


QGIS version | 3.12.0-București | QGIS code revision | cd141490ec
-- | -- | -- | --
Compiled against Qt | 5.11.2 | Running against Qt | 5.11.2
Compiled against GDAL/OGR | 3.0.4 | Running against GDAL/OGR | 3.0.4
Compiled against GEOS | 3.8.0-CAPI-1.13.1 | Running against GEOS | 3.8.0-CAPI-1.13.1
Compiled against SQLite | 3.29.0 | Running against SQLite | 3.29.0
PostgreSQL Client Version | 11.5 | SpatiaLite Version | 4.3.0
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.8
Compiled against PROJ | 6.3.1 | Running against PROJ | Rel. 6.3.1, February 10th, 2020
OS Version | Windows 10 (10.0)
Active python plugins | DataPlotly; GroupStats; networks; OSMDownloader; qgis2web; QNEAT3; QuickOSM; quick_map_services; db_manager; processing

**Comment by nyalldawson** [comment]
I can't reproduce on Linux

**Comment by josephholler** [comment]
I'm starting to question my sanity with this problem. I've seen it occur on 6 different Windows 10 computers with 4 different users in 3.10 and 3.12, but at this moment I'm failing to reproduce it on the same system I used earlier today (metadata below). If you have any ideas about factors I'm not thinking of, please share! It cannot possibly be a random issue?

QGIS version | 3.10.2-A Coruña | QGIS code revision | d4cd3cfe5a
-- | -- | -- | --
Compiled against Qt | 5.11.2 | Running against Qt | 5.11.2
Compiled against GDAL/OGR | 3.0.3 | Running against GDAL/OGR | 3.0.3
Compiled against GEOS | 3.8.0-CAPI-1.13.1 | Running against GEOS | 3.8.0-CAPI-1.13.1
Compiled against SQLite | 3.29.0 | Running against SQLite | 3.29.0
PostgreSQL Client Version | 11.5 | SpatiaLite Version | 4.3.0
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.8
Compiled against PROJ | 6.3.0 | Running against PROJ | Rel. 6.3.0, January 1st, 2020
OS Version | Windows 10 (10.0)
Active python plugins | db_manager; MetaSearch; processing

**Comment by gioman** [comment]
> I can't reproduce on Linux

@josephholler can't replicate on 3.12.0 on Windows either.

**Comment by josephholler** [comment]
I'm repeatedly and consistently seeing the problem again today while students work on this data for a lab using 3.10.2 . I'm attaching an exact copy of their data. 
[c_wkend02.zip](https://github.com/qgis/QGIS/files/4245881/c_wkend02.zip)

**Comment by josephholler** [comment]
I'm really struggling with this problem. Our labs have 3.10.2 installed on Windows 10 machines and I'd say about two thirds of students are getting output layers with missing CRS metadata and the other one third or so are ok. I cannot find any pattern to explain the difference. I was seeing it again with EPSG 6454 today.

**Comment by nyalldawson** [comment]
You should test with 3.10.3 - there's been a lot of projection related fixes in that release

**Comment by GCouet** [comment]
I've had a similar experience with 3.12.0 on Windows.

Simply trying to save a layer (that has a CRS set) to an existing geopackage seems to strip it of the CRS.
I went ahead and did the same process in my 3.8.2 install with no issue. The layer is correctly added with a CRS to the geopackage.

**Comment by gioman** [comment]
> I've had a similar experience with 3.12.0 on Windows.

@GCouet can you attach your sample data and *precise* steps to follow to replicate? thanks!

**Comment by GCouet** [comment]
Sorry for the late reply, but here is my experimentation. The probem seems to randomly fix itself, but I haven't figured out what I did that might have fixed the first geopackage. To my knowledge I did nothing different for the two sample files. The first one is the one that now works. The second one is the one that remains broken.

[mino.zip](https://github.com/qgis/QGIS/files/4287578/mino.zip)

- In the gpkg:
  - _mino_ layer: original layer with correct CRS from before 3.12
  - _Points_ layer: created as a scratch layer (with EPSG:32189 as a CRS), **made permanent** into the gpkg
  - _Lake_ layer: with _mino_ loaded in QGIS, **saved feature as** into the gpkg.

Both new layers in the gpkg have no CRS assigned. Doing those same steps in 3.8.2 yielded layers with an assigned CRS. Additional test:

- I took the same _Points_ scratch layer and **made permanent** into a **new** gpkg, the CRS was correctly assigned.

Strangely, after doing these tasks, mino.gpkg now works fine, which is the uploaded one above.

I tried with another gpkg (below) that was created before 3.12 and the problem happened again. I did the same steps but I also tried saving repeatedly into it, deleting layers in the browser. Saving the layer into a new gpkg yields a layer with no CRS, contrary to the earlier gpkg. Here is the file:

[line_scratch.zip](https://github.com/qgis/QGIS/files/4287720/line_scratch.zip)


QGIS version | 3.12.0-București | QGIS code revision | cd141490ec
-- | -- | -- | --
Compiled against Qt | 5.11.2 | Running against Qt | 5.11.2
Compiled against GDAL/OGR | 3.0.4 | Running against GDAL/OGR | 3.0.4
Compiled against GEOS | 3.8.0-CAPI-1.13.1 | Running against GEOS | 3.8.0-CAPI-1.13.1
Compiled against SQLite | 3.29.0 | Running against SQLite | 3.29.0
PostgreSQL Client Version | 11.5 | SpatiaLite Version | 4.3.0
QWT Version | 6.1.3 | QScintilla2 Version | 2.10.8
Compiled against PROJ | 6.3.1 | Running against PROJ | Rel. 6.3.1, February 10th, 2020
OS Version | Windows 10 (10.0)
Active python plugins | db_manager; MetaSearch; processing
